### PR TITLE
added signup page

### PR DESCRIPTION
**Pull request by paulmuia**
created signup page

### CODE REVIEW

File: lib/pages/signup page.dart [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+                                  focusedBorder: OutlineInputBorder(
+                                      borderRadius: BorderRadius.circular(15),
+                                      borderSide: const BorderSide(
+                                        width: 1.0,
+                                        color: Color.fromARGB(255, 2, 35, 63),
+                                      )
+                                  ),
+                                  enabledBorder: OutlineInputBorder(
+                                      borderRadius: BorderRadius.circular(15),
+                                      borderSide: const BorderSide(
```
Comment (Piexie3) [inline lib/pages/signup page.dart]: Create a Util class for all the text fields

File: lib/pages/signup page.dart [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+                                  prefixIcon: const Icon(Icons.person,color: Color.fromARGB(255, 2, 35, 63),),
+                                  focusedBorder: OutlineInputBorder(
+                                      borderRadius: BorderRadius.circular(15),
+                                      borderSide: const BorderSide(
+                                        width: 1.0,
+                                        color: Color.fromARGB(255, 2, 35, 63),
+                                      )
+                                  ),
+                                  enabledBorder: OutlineInputBorder(
+                                      borderRadius: BorderRadius.circular(15),
```
Comment (Piexie3) [inline lib/pages/signup page.dart]: create one Util class for text field to minimise repetition

**Review by Piexie3 — CHANGES_REQUESTED** [review]
create a stateless widget for this class if you will not need to rebuilt the widget or using setState() anywhere

File: lib/pages/signup page.dart (hunk 2 above)
Comment (paulmuia) [reply lib/pages/signup page.dart]: created a util class for all textfields and imlimented it in the signup page

**Comment by paulmuia** [comment]
created a util class for all textfields and imlimented it in the signup page

File: lib/pages/signup page.dart [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+class SignUpPage extends StatefulWidget {
+  const SignUpPage({super.key});
+
+  @override
+  State<SignUpPage> createState() => _SignUpPageState();
+}
+
+class _SignUpPageState extends State<SignUpPage> {
+  @override
+  Widget build(BuildContext context) {
```
Comment (Piexie3) [inline lib/pages/signup page.dart:27]:
```suggestion
  Widget build(BuildContext context) {
  // initialize your controllers here
 final TextEditingController firstNameController = TextEditingController();
final TextEditingController lastNameController = TextEditingController();
final TextEditingController idController = TextEditingController();
```

File: lib/pages/signup page.dart [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+                    ),
+                    const SizedBox(
+                      height: 20,
+                    ),
+                    Padding(
+                        padding: const EdgeInsets.symmetric(horizontal: 20),
+                        child: TextFieldForm(
+                            hintText: 'Enter first name',
+                            prefix: const Icon(Icons.person),
+                            controller: TextEditingController(),
```
Comment (Piexie3) [inline lib/pages/signup page.dart]:
```suggestion
                            controller:  firstNameController ,
```

File: lib/pages/signup page.dart [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+                            keyboardType: TextInputType.name)),
+                    const SizedBox(
+                      height: 10,
+                    ),
+                    Padding(
+                        padding: const EdgeInsets.symmetric(horizontal: 20),
+                        child: TextFieldForm(
+                            hintText: 'Enter ID number',
+                            prefix: const Icon(Icons.person),
+                            controller: TextEditingController(),
```
Comment (Piexie3) [inline lib/pages/signup page.dart]: make sure to change all the controllers

```suggestion
                            controller: lastNameController,
```

**Review by Piexie3 — CHANGES_REQUESTED** [review]

**Comment by paulmuia** [comment]
modified controllers

**Comment by paulmuia** [comment]
added login page

**Review by Piexie3 — APPROVED** [review]

**Pull request merged**